pandas/core/indexes/datetimes.py

```diff
@@ -1633,8 +1643,7 @@ def date(self):
         Returns numpy array of python datetime.date objects (namely, the date
         part of Timestamps without timezone information).
         """
-        return self._maybe_mask_results(libalgos.arrmap_object(
-            self.asobject.values, lambda x: x.date()))
+        return self.normalize().to_pydate()
```

Comment:
just call the code for what you have in ``to_pydate()`` above, and remove that method; we don't actually want to expose this to the user.